### PR TITLE
DAST Scanning

### PR DESCRIPTION
**Pull request by jorg3lopez**
# Dynamic Application Security Test 

This PR adds DAST scanning to our application. The scanning is done as part of a workflow. The event gets triggered when pushing to the main branch. 
Currently, the workflow fires up when pushing to the remote branch, this was done on purpose in order to help with testing. Once the PR gets approved, the DAST scanning will get triggered every time something gets pushed to the main branch.

## Changes
New Files
- health-check.sh
- dast.yml

health-check.sh - Checks to see if the api is functional by accessing the health endpoint.

dast.yml - Workflow that uses OWASP ZAP for DAST scanning

## Testing
You want to test the script and also the workflow.

### Testing the script:
- run shadowJar: `./gradlew shadowJar`
- build docker container: `docker compose up --build -d`
- run the script: `./health-check.sh`

### Testing the workflow:
The easiest way to do this would be to add a minor change to the branch and then push it.
Check the Actions on github and make sure everything passes.
There should be an artifact that contains warnings such as 404 status codes. 

## Issue
#77.

### CODE REVIEW

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java
Comment (jorg3lopez) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java:25]: I think we can refactor this better. Maybe we can create a method that adds all of the existing documentation. It feels like `getBaselineDocumentation()` and `getEtorDocumentation()` do a similar job. What are your thoughts?

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java
Comment (jorg3lopez) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:32]: Better naming suggestions are welcome.

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java
Comment (jorg3lopez) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java:52]: This is giving me a code smell for the Sonar scan

File: .github/workflows/dast.yml
Comment (jorg3lopez) [inline .github/workflows/dast.yml:5]: This will change once it gets approved, it should be triggered when pushed to master. Other option can be having it run on every PR; thoughts?

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java
Comment (jorg3lopez) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:32]: endpoints/ endpointsHandlers?

File: health-check.sh
Comment (jorg3lopez) [inline health-check.sh:24]: Remove "if container is running"

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java
Comment (jorg3lopez) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java]: Did not need this line of code as the Etor documentation comes in via the parameter `openApiSpecifications`

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java
Comment (jorg3lopez) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java]: renamed to `endpoints`

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java
Comment (jorg3lopez) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/OpenApi.java]: We don't need this code as the etor documentation comes in through the method parameter

File: health-check.sh
Comment (jorg3lopez) [reply health-check.sh]: Done. This will make the script more general and it can be use with other events

File: health-check.sh
Comment (halprin) [inline health-check.sh:37]:
```suggestion
        echo "Status Code: $HTTP_CODE"
```

File: health-check.sh
Comment (halprin) [inline health-check.sh:11]: I don't think this variable is being used.

File: .github/workflows/dast.yml
Comment (halprin) [reply .github/workflows/dast.yml:5]: I think it should run on every PR and every push to `main`.

Comment (jorg3lopez) [reply .github/workflows/dast.yml]: Good call.

File: health-check.sh
Comment (jorg3lopez) [reply health-check.sh]: Good catch, thanks!

File: health-check.sh
Comment (jorg3lopez) [reply health-check.sh]: You are right, this was part of the whole checking for the container; great catch!

File: .github/workflows/dast.yml
Comment (jorg3lopez) [reply .github/workflows/dast.yml:7]: Great idea!

**Review by halprin — APPROVED** [review]

**Pull request merged**
